app/assets/javascripts/components/general_button.es6.jsx

```diff
@@ -43,12 +57,31 @@ class GeneralButton extends React.Component {
     }
   }
 
+  handleMouseDown(event) {
+    this.setState({ hover: true });
+  }
+
+  handleMouseUp(event) {
+    this.setState({ hover: false });
+  }
+
+  handleMouseLeave(event) {
+    if (this.state.hover) {
+      this.setState({ hover: false });
+    }
+  }
+
   render() {
+    var style = Object.assign(
+      this.styles.container,
+      this.state.hover && this.styles.hover
+    );
```

Comment:
how does this style Object work? A little confused on how the second parameter works. I can see why the hover state can act as a boolean, not really sure about 'this.styles.hover'